File: frontend/types/knowledge.ts
export type DocumentStatus = "indexed" | "failed" | "missing";

export interface KnowledgeDocument {
  id: string;
  source_path: string;
  file_name: string;
  file_extension: string;
  mime_type: string;
  file_size: number;
  status: DocumentStatus;
  error_message: string | null;
  created_at: string;
  updated_at: string;
  indexed_at: string | null;
}

export interface KnowledgeDocumentList {
  items: KnowledgeDocument[];
  page: number;
  page_size: number;
  total: number;
}

export interface KnowledgeScanResult {
  discovered: number;
  indexed: number;
  unchanged: number;
  unsupported: number;
  failed: number;
}

export interface KnowledgeSearchResult {
  document_id: string;
  file_name: string;
  source_path: string;
  source_locator: string;
  excerpt: string;
  relevance_score: number | null;
}

export interface KnowledgeSearchResponse {
  query: string;
  items: KnowledgeSearchResult[];
}

File: backend/app/schemas/knowledge.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, Field


DocumentStatus = Literal["indexed", "failed", "missing"]


class ScanKnowledgeResponse(BaseModel):
    discovered: int
    indexed: int
    unchanged: int
    unsupported: int
    failed: int


class DocumentSummaryResponse(BaseModel):
    id: UUID
    source_path: str
    file_name: str
    file_extension: str
    mime_type: str
    file_size: int
    status: DocumentStatus
    error_message: str | None
    created_at: datetime
    updated_at: datetime
    indexed_at: datetime | None


class DocumentListResponse(BaseModel):
    items: list[DocumentSummaryResponse]
    page: int = Field(ge=1)
    page_size: int = Field(ge=1)
    total: int = Field(ge=0)


class DocumentChunkSummary(BaseModel):
    chunk_index: int
    source_locator: str


class DocumentDetailResponse(DocumentSummaryResponse):
    chunk_count: int
    chunks: list[DocumentChunkSummary]


class DeleteDocumentResponse(BaseModel):
    document_id: UUID


class KnowledgeSearchResult(BaseModel):
    document_id: UUID
    file_name: str
    source_path: str
    source_locator: str
    excerpt: str
    relevance_score: float | None = None


class KnowledgeSearchResponse(BaseModel):
    items: list[KnowledgeSearchResult]
    query: str
